src/send_future.rs

```diff
@@ -0,0 +1,180 @@
+use crate::manager::AddressMessage;
+use crate::receiver::Receiver;
+use crate::Disconnected;
+use flume::r#async::SendFut;
+use futures_core::future::BoxFuture;
+use futures_util::FutureExt;
+use std::future::Future;
+use std::marker::PhantomData;
+use std::mem;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+/// A [`Future`] that represents the state of sending a message to an actor.
+///
+/// By default, a `SendFuture` will also wait until the handler has finished executing and resolve to the return value (see [`Handler::Return`](crate::Handler::Return)).
+/// This behaviour can be changed by calling [`recv_async`](SendFuture::recv_async).
+///
+/// When toggled to receive the return value asynchronously, this future will resolve once the message is successfully queued into the actor's mailbox. If the actors mailbox is bounded, this future will yield `Pending` until there is space in the mailbox. This allows an actor to exercise backpressure on its users.
+#[must_use = "Futures do nothing unless polled"]
+pub struct SendFuture<R, F, TRecvSyncMarker> {
+    inner: SendFutureInner<R, F>,
+    phantom: PhantomData<TRecvSyncMarker>,
+}
+
+/// Marker type for receiving the return value of a [`Handler`](crate::Handler) **synchronously**.
+pub enum ReceiveSync {}
```

Comment:
When you say synchronously, I assume you mean as part of the same future as the `send()`, so still not with blocking channel `recv`?

Reply:
Yes, synchronously here means within the same future/task!

Reply:
I think we should document `xtra`-wide that we attempt to never block threads whatsoever so "blocking" always just means "blocking the task".